cli/flox/doc/manifest.toml.md

```diff
@@ -252,15 +252,48 @@ DB_URL = "http://localhost:2000"
 SERVER_PORT = "3000"
 ```
 
+## `[profile]`
+
+The `[profile]` section of the manifest allows you to specify scripts that will
+be sourced by your shell immediately after activating the environment.
+These scripts can be used to perform additional setup for a consumer of the
+environment e.g. spawning a background process, dynamically setting environment
+variables, or creating files and directories.
+The environment variables declared in the `[vars]` section can be used in the
+`profile` scripts.
+
+```toml
+[profile]
+common = """
+    export SERVER_URL="http://localhost:$SERVER_PORT"
+"""
+
+bash = """
+    export MYSHELL="bash"
+"""
+
+zsh = """
+    export MYSHELL="zsh"
+"""
+```
+
+The `profile.common` script is intended to be common setup that can be sourced
+by any shell,
+but it is your responsibility to make sure that the script is compatible with
+any shells that may consume the environment.
+The `profile.bash` and `profile.zsh` scripts are sourced *after* the
+`profile.common` script,
+and are only sourced by the corresponding shell.
+The shell-specific profile scripts are intended to contain any shell functions, 
+aliases, variables, etc that could be specific to a user's shell.
+
 ## `[hook]`
 
 The `[hook]` section of the manifest allows you to specify a script that's
-executed immediately after the environment is activated.
-Since the hook runs after activation, the environment variables in the `[vars]`
-section may be referenced within the hook.
-
-Common usages for environment hooks are printing usage messages or performing
-setup operations such as initializing a database or starting a server.
+*executed* after the `profile` scripts have been sourced.
+Since hooks run after activation
+the environment variables in the `[vars]` section may be referenced within the
+hook.
 
 The `hook.script` and `hook.on-activate` options are mutually exclusive.
```

Comment:
It's not clear what the difference between `hook.on-activate` and `hook.script` is.

Reply:
I understand now... updated the docs a little.